test/sim/abilities/protean.js

```diff
@@ -28,18 +32,90 @@ describe('Protean', function () {
 		//Turn 1 of Bounce
 		battle.makeChoices();
-		assert.equal(wynaut.getTypes().join('/'), 'Flying');
+		assert(wynaut.hasType('Flying'));
 
 		//Turn 2 of Bounce
 		battle.makeChoices();
-		assert.equal(wynaut.getTypes().join('/'), 'Flying');
+		assert(wynaut.hasType('Flying'));
+	});
+
+	it(`should not change the user's type when using moves that fail earlier than Protean will activate`, function () {
+		battle = common.createBattle([[
+			{species: "Kecleon", ability: 'protean', moves: ['fling', 'suckerpunch', 'steelroller', 'aurawheel']},
+			{species: "Kecleon", ability: 'protean', moves: ['magnetrise', 'ingrain', 'burnup', 'auroraveil']},
+		], [
+			{species: "Wynaut", moves: ['sleeptalk']},
+		]]);
+
+		let kecleon = battle.p1.active[0];
+
+		// Fling with no item
+		battle.makeChoices('move fling', 'move sleeptalk');
+		assert(kecleon.hasType('Normal'), "Fling changed the user's type when it should not have.");
+
+		// Sucker Punch into status move
+		battle.makeChoices('move suckerpunch', 'move sleeptalk');
+		assert(kecleon.hasType('Normal'), "Sucker Punch changed the user's type when it should not have.");
+
+		// Steel Roller with no Terrain active
+		battle.makeChoices('move steelroller', 'move sleeptalk');
+		assert(kecleon.hasType('Normal'), "Steel Roller changed the user's type when it should not have.");
+
+		// Aura Wheel when user is not Morpeko
+		battle.makeChoices('move aurawheel', 'move sleeptalk');
+		assert(kecleon.hasType('Normal'), "Aura Wheel changed the user's type when it should not have.");
+
+		battle.makeChoices('switch 2', 'move sleeptalk');
+		kecleon = battle.p1.active[0];
+
+		// Burn Up when user is not Fire-type
+		battle.makeChoices('move burnup', 'move sleeptalk');
+		assert(kecleon.hasType('Normal'), "Burn Up changed the user's type when it should not have.");
+
+		// Aurora Veil when hail is not active
+		battle.makeChoices('move auroraveil', 'move sleeptalk');
+		assert(kecleon.hasType('Normal'), "Aurora Veil changed the user's type when it should not have.");
+
+		// Magnet Rise when user has Ingrain; note that Kecleon becomes Grass-type from Ingrain
+		battle.makeChoices('move ingrain', 'move sleeptalk');
+		battle.makeChoices('move magnetrise', 'move sleeptalk');
+		assert(kecleon.hasType('Grass'), "Magnet Rise changed the user's type when it should not have.");
+
+		// More examples: https://www.smogon.com/forums/threads/sword-shield-battle-mechanics-research.3655528/post-8548957
 	});
 
-	it('should not change the user\'s type when using Fling with no item', function () {
-		battle = common.createBattle();
-		battle.setPlayer('p1', {team: [{species: "Kecleon", ability: 'protean', moves: ['fling']}]});
-		battle.setPlayer('p2', {team: [{species: "Wobbuffet", moves: ['counter']}]});
-		battle.makeChoices('move fling', 'move counter');
-		assert(battle.p1.active[0].hasType('Normal'));
+	it(`should not change the user's type when abilities that fail earlier than Protean will activate`, function () {
```

Comment:
This is worded strangely. I would change it to something like:
```suggestion
	it(`should not change the user's type when abilities that activate earlier than Protean cause user's move to fail`, function () {
```

Reply:
Good idea!